[sdk/python] `make lint` not running in CI?

Running `make lint` locally in `sdk/python` was resulting in:

```
LINT:
pipenv run mypy ./lib/pulumi --config-file=mypy.ini
lib/pulumi/x/automation/_minimum_version.py:15: error: Skipping analyzing 'semver': found module but no type hints or library stubs
lib/pulumi/x/automation/_cmd.py:62: error: Item "None" of "Optional[IO[bytes]]" has no attribute "readline"
lib/pulumi/runtime/rpc.py:118: error: Name 'CustomResource' is not defined
lib/pulumi/runtime/resource.py:524: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")
lib/pulumi/runtime/resource.py:526: error: Incompatible types in assignment (expression has type "Optional[Dict[str, PropertyDependencies]]", variable has type "Dict[str, PropertyDependencies]")
lib/pulumi/output.py:393: error: Name 'all_outputs' already defined on line 391
lib/pulumi/output.py:400: error: Argument 1 to "get_resources" has incompatible type "object"; expected "List[Any]"
lib/pulumi/output.py:401: error: Argument 1 to "is_known" has incompatible type "object"; expected "List[Any]"
lib/pulumi/output.py:402: error: Argument 1 to "is_secret" has incompatible type "object"; expected "List[Any]"
lib/pulumi/x/automation/_local_workspace.py:20: error: Skipping analyzing 'semver': found module but no type hints or library stubs
lib/pulumi/x/automation/_local_workspace.py:20: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 10 errors in 6 files (checked 58 source files)
make: *** [lint] Error 1
```

https://github.com/pulumi/pulumi/pull/6627 fixes these issues.

Are we not running the lint target in CI?

--- a/.github/workflows/run-build-and-acceptance-tests.yml
+++ b/.github/workflows/run-build-and-acceptance-tests.yml
@@ -30,7 +30,7 @@ jobs:
             Please view the results of the PR Build + Acceptance Tests Run [Here][1]
 
             [1]: ${{ steps.vars.outputs.run-url }}
-  lint:
+  go-lint:
     container: golangci/golangci-lint:latest
     name: Lint ${{ matrix.directory }}
     strategy:
@@ -43,6 +43,60 @@ jobs:
       - name: Lint ${{ matrix.directory }}
         run: |
           cd ${{ matrix.directory }} && golangci-lint run -c ../.golangci.yml
+  sdk-lint:
+    name: Lint SDKs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Set up DotNet ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+      - run: dotnet nuget add source ${{ runner.temp }}/opt/pulumi/nuget
+      - name: Set up Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install pipenv
+        uses: dschep/install-pipenv-action@v1
+      - name: Setup git
+        run: |
+          git config --global user.email "you@example.com"
+          git config --global user.name "Your Name"
+      - name: Update path
+        run: |
+          echo "${{ runner.temp }}/opt/pulumi/bin" >> $GITHUB_PATH
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Fetch Tags
+        run: |
+          git fetch --quiet --prune --unshallow --tags
+      - name: Set Go Dep path
+        run: |
+          echo "PULUMI_GO_DEP_ROOT=$(dirname $(pwd))" >> $GITHUB_ENV
+      - name: Install pulumictl
+        uses: jaxxstorm/action-install-gh-release@v1.2.0
+        with:
+          repo: pulumi/pulumictl
+      - name: Ensure
+        run: |
+          make ensure
+      - name: Lint Node
+        run: |
+          cd sdk/nodejs && make lint
+      - name: Lint Python
+        run: |
+          cd sdk/python && make lint
+      - name: Lint .NET
+        run: |
+          cd sdk/dotnet && make lint
   build_and_test:
     name: Build & Test
     strategy:

--- a/sdk/nodejs/output.ts
+++ b/sdk/nodejs/output.ts
@@ -16,6 +16,8 @@ import { Resource } from "./resource";
 import * as runtime from "./runtime";
 import * as utils from "./utils";
 
+/*tslint:disable:no-shadowed-variable*/
+
 /**
  * Output helps encode the relationship between Resources in a Pulumi application. Specifically an
  * Output holds onto a piece of Data and the Resource it was generated from. An Output value can

--- a/sdk/nodejs/runtime/config.ts
+++ b/sdk/nodejs/runtime/config.ts
@@ -42,7 +42,7 @@ export function setAllConfig(c: {[key: string]: string}) {
 export function setConfig(k: string, v: string): void {
     const config = parseConfig();
     config[cleanKey(k)] = v;
-    persistConfig(config)
+    persistConfig(config);
 }
 
 /**

--- a/sdk/nodejs/runtime/rpc.ts
+++ b/sdk/nodejs/runtime/rpc.ts
@@ -642,7 +642,7 @@ export function register<T extends { readonly version?: string }>(source: Map<st
 
     log.debug(`registering ${registrationType} ${key}@${item.version}`);
     items.push(item);
-    return true
+    return true;
 }
 
 /** @internal */

--- a/sdk/nodejs/tests/runtime/settings.spec.ts
+++ b/sdk/nodejs/tests/runtime/settings.spec.ts
@@ -24,7 +24,7 @@ describe("settings", () => {
     after(()=> {
         runtime._reset();
         runtime.setAllConfig({});
-    })
+    });
     it("runtime options and config are read from and updated via the environment", () => {
         runtime._setTestModeEnabled(false);
 
@@ -38,7 +38,7 @@ describe("settings", () => {
         runtime._setQueryMode(isQueryMode);
         const key = "k";
         const val = "v";
-        runtime.setConfig(key, val)
+        runtime.setConfig(key, val);
 
         assert.strictEqual(runtime.getProject(), testProject);
         assert.strictEqual(runtime.getStack(), testStack);
@@ -46,10 +46,10 @@ describe("settings", () => {
         assert.strictEqual(runtime.isQueryMode(), isQueryMode);
         assert.strictEqual(runtime.getConfig(key), val);
 
-        assert.strictEqual(testProject, process.env["PULUMI_NODEJS_PROJECT"])
-        assert.strictEqual(testStack, process.env["PULUMI_NODEJS_STACK"])
-        assert.strictEqual("true", process.env["PULUMI_NODEJS_DRY_RUN"])
-        assert.strictEqual("true", process.env["PULUMI_NODEJS_QUERY_MODE"])
-        assert.strictEqual(JSON.stringify({[key]: val}), process.env["PULUMI_CONFIG"])
+        assert.strictEqual(testProject, process.env["PULUMI_NODEJS_PROJECT"]);
+        assert.strictEqual(testStack, process.env["PULUMI_NODEJS_STACK"]);
+        assert.strictEqual("true", process.env["PULUMI_NODEJS_DRY_RUN"]);
+        assert.strictEqual("true", process.env["PULUMI_NODEJS_QUERY_MODE"]);
+        assert.strictEqual(JSON.stringify({[key]: val}), process.env["PULUMI_CONFIG"]);
     });
 });

--- a/sdk/nodejs/x/automation/server.ts
+++ b/sdk/nodejs/x/automation/server.ts
@@ -103,7 +103,7 @@ export class LanguageServer<T> implements grpc.UntypedServiceImplementation {
         } catch (e) {
             const err = e instanceof Error ? e : new Error(`unknown error ${e}`);
             resp.setError(err.message);
-            callback(err, undefined)
+            callback(err, undefined);
         }
 
         callback(undefined, resp);
@@ -133,7 +133,7 @@ function newUncaughtHandler(errorSet: Set<Error>): (err: Error) => void {
         //
         // If both the stack and message are empty, then just stringify the err object itself. This
         // is also necessary as users can throw arbitrary things in JS (including non-Errors).
-        let defaultMessage = ""
+        let defaultMessage = "";
         if (!!err) {
             defaultMessage = err.stack || err.message || ("" + err);
         }

--- a/sdk/python/lib/pulumi/config.py
+++ b/sdk/python/lib/pulumi/config.py
@@ -122,8 +122,8 @@ class Config:
             return None
         try:
             return int(v)
-        except:
-            raise ConfigTypeError(self.full_key(key), v, 'int')
+        except Exception as e:
+            raise ConfigTypeError(self.full_key(key), v, 'int') from e
 
     def get_secret_int(self, key: str) -> Optional[Output[int]]:
         """
@@ -156,8 +156,8 @@ class Config:
             return None
         try:
             return float(v)
-        except:
-            raise ConfigTypeError(self.full_key(key), v, 'float')
+        except Exception as e:
+            raise ConfigTypeError(self.full_key(key), v, 'float') from e
 
     def get_secret_float(self, key: str) -> Optional[Output[float]]:
         """
@@ -186,8 +186,8 @@ class Config:
             return None
         try:
             return json.loads(v)
-        except:
-            raise ConfigTypeError(self.full_key(key), v, "JSON object")
+        except Exception as e:
+            raise ConfigTypeError(self.full_key(key), v, "JSON object") from e
 
     def get_secret_object(self, key: str) -> Optional[Output[Any]]:
         """
@@ -366,7 +366,7 @@ class ConfigTypeError(errors.RunError):
         self.key = key
         self.value = value
         self.expect_type = expect_type
-        super(ConfigTypeError, self).__init__(
+        super().__init__(
             "Configuration '%s' value '%s' is not a valid '%s'" % (key, value, expect_type))
 
 
@@ -382,6 +382,6 @@ class ConfigMissingError(errors.RunError):
 
     def __init__(self, key: str) -> None:
         self.key = key
-        super(ConfigMissingError, self).__init__(
+        super().__init__(
             "Missing required configuration variable '%s'\n" % key +
             "    please set a value using the command `pulumi config set %s <value>`" % key)

--- a/sdk/python/lib/pulumi/dynamic/dynamic.py
+++ b/sdk/python/lib/pulumi/dynamic/dynamic.py
@@ -253,4 +253,4 @@ class Resource(CustomResource):
         props = cast(dict, props)
         props[PROVIDER_KEY] = serialize_provider(provider)
 
-        super(Resource, self).__init__("pulumi-python:dynamic:Resource", name, props, opts)
+        super().__init__("pulumi-python:dynamic:Resource", name, props, opts)

--- a/sdk/python/lib/pulumi/output.py
+++ b/sdk/python/lib/pulumi/output.py
@@ -256,7 +256,9 @@ class Output(Generic[T]):
             # Since Output.all works on lists early, serialize the class's __dict__ into a list of lists first.
             # Once we have a output of the list of properties, we can use an apply to re-hydrate it back as an instance.
             items = [[k, Output.from_input(v)] for k, v in val.__dict__.items()]
-            fn = cast(Callable[[List[Any]], T], lambda props: typ(**{k: v for k, v in props}))  # type: ignore
+
+            # pylint: disable=unnecessary-comprehension
+            fn = cast(Callable[[List[Any]], T], lambda props: typ(**{k: v for k, v in props})) # type: ignore
             return Output.all(*items).apply(fn, True)
 
         # Is a dict or list? Recurse into the values within them.

--- a/sdk/python/lib/pulumi/runtime/resource.py
+++ b/sdk/python/lib/pulumi/runtime/resource.py
@@ -190,6 +190,7 @@ def get_resource(res: 'Resource', props: 'Inputs', custom: bool, urn: str) -> No
                     return monitor.Invoke(req)
                 except grpc.RpcError as exn:
                     handle_grpc_error(exn)
+                    return None
 
             resp = await asyncio.get_event_loop().run_in_executor(None, do_invoke)
 
@@ -298,6 +299,7 @@ def read_resource(res: 'CustomResource', ty: str, name: str, props: 'Inputs', op
                     return monitor.ReadResource(req)
                 except grpc.RpcError as exn:
                     handle_grpc_error(exn)
+                    return None
 
             resp = await asyncio.get_event_loop().run_in_executor(None, do_rpc_call)
 
@@ -437,6 +439,7 @@ def register_resource(res: 'Resource',
                     return monitor.RegisterResource(req)
                 except grpc.RpcError as exn:
                     handle_grpc_error(exn)
+                    return None
 
             resp = await asyncio.get_event_loop().run_in_executor(None, do_rpc_call)
         except Exception as exn:
@@ -492,6 +495,7 @@ def register_resource_outputs(res: 'Resource', outputs: 'Union[Inputs, Output[In
                 return monitor.RegisterResourceOutputs(req)
             except grpc.RpcError as exn:
                 handle_grpc_error(exn)
+                return None
 
         await asyncio.get_event_loop().run_in_executor(None, do_rpc_call)
         log.debug(

--- a/sdk/python/lib/pulumi/runtime/rpc.py
+++ b/sdk/python/lib/pulumi/runtime/rpc.py
@@ -319,8 +319,8 @@ def deserialize_resource(ref_struct: struct_pb2.Struct, keep_unknowns: Optional[
 
     # If we've made it here, deserialize the reference as either a URN or an ID (if present).
     if "id" in ref_struct:
-        id = ref_struct["id"]
-        return deserialize_property(UNKNOWN if id == "" else id, keep_unknowns)
+        ref_id = ref_struct["id"]
+        return deserialize_property(UNKNOWN if ref_id == "" else ref_id, keep_unknowns)
 
     return urn
 
@@ -563,12 +563,12 @@ def contains_unknowns(val: Any) -> bool:
         if known_types.is_unknown(val):
             return True
 
-        if not any([x is val for x in stack]):
+        if not any((x is val for x in stack)):
             stack.append(val)
             if isinstance(val, dict):
-                return any([impl(val[k], stack) for k in val])
+                return any((impl(val[k], stack) for k in val))
             if isinstance(val, list):
-                return any([impl(x, stack) for x in val])
+                return any((impl(x, stack) for x in val))
         return False
 
     return impl(val, [])

--- a/sdk/python/lib/pulumi/runtime/settings.py
+++ b/sdk/python/lib/pulumi/runtime/settings.py
@@ -219,9 +219,9 @@ async def monitor_supports_feature(feature: str) -> bool:
                 resp = monitor.SupportsFeature(req)
                 return resp.hasSupport
             except grpc.RpcError as exn:
-                if exn.code() == grpc.StatusCode.UNIMPLEMENTED:
-                    return False
-                handle_grpc_error(exn)
+                if exn.code() != grpc.StatusCode.UNIMPLEMENTED: # pylint: disable=no-member
+                    handle_grpc_error(exn)
+                return False
 
         result = await asyncio.get_event_loop().run_in_executor(None, do_rpc_call)
         SETTINGS.feature_support[feature] = result

--- a/sdk/python/lib/pulumi/runtime/stack.py
+++ b/sdk/python/lib/pulumi/runtime/stack.py
@@ -118,7 +118,7 @@ class Stack(ComponentResource):
 
         # Now invoke the registration to begin creating this resource.
         name = '%s-%s' % (get_project(), get_stack())
-        super(Stack, self).__init__('pulumi:pulumi:Stack', name, None, None)
+        super().__init__('pulumi:pulumi:Stack', name, None, None)
 
         # Invoke the function while this stack is active and then register its outputs.
         self.outputs = dict()

--- a/sdk/python/lib/pulumi/x/automation/__init__.py
+++ b/sdk/python/lib/pulumi/x/automation/__init__.py
@@ -99,6 +99,7 @@ from ._config import (
     ConfigValue
 )
 
+# pylint: disable=redefined-builtin
 from .errors import (
     StackNotFoundError,
     StackAlreadyExistsError,

--- a/sdk/python/lib/pulumi/x/automation/_local_workspace.py
+++ b/sdk/python/lib/pulumi/x/automation/_local_workspace.py
@@ -15,10 +15,10 @@
 import os
 import tempfile
 import json
-import yaml
 from datetime import datetime
-from semver import VersionInfo
 from typing import Optional, List, Mapping, Callable
+from semver import VersionInfo
+import yaml
 
 from ._config import ConfigMap, ConfigValue
 from ._project_settings import ProjectSettings
@@ -112,8 +112,10 @@ class LocalWorkspace(Workspace):
         path = os.path.join(self.work_dir, f"Pulumi{found_ext}")
         writable_settings = {key: settings.__dict__[key] for key in settings.__dict__ if settings.__dict__[key] is not None}
         with open(path, "w") as file:
-            json.dump(writable_settings, file, indent=4) if found_ext == ".json" else yaml.dump(
-                writable_settings, stream=file)
+            if found_ext == ".json":
+                json.dump(writable_settings, file, indent=4)
+            else:
+                yaml.dump(writable_settings, stream=file)
 
     def stack_settings(self, stack_name: str) -> StackSettings:
         stack_settings_name = get_stack_settings_name(stack_name)
@@ -140,8 +142,10 @@ class LocalWorkspace(Workspace):
                 break
         path = os.path.join(self.work_dir, f"Pulumi.{stack_settings_name}{found_ext}")
         with open(path, "w") as file:
-            json.dump(settings.__dict__, file, indent=4) if found_ext == ".json" else yaml.dump(
-                settings.__dict__, stream=file)
+            if found_ext == ".json":
+                json.dump(settings.__dict__, file, indent=4)
+            else:
+                yaml.dump(settings.__dict__, stream=file)
 
     def serialize_args_for_op(self, stack_name: str) -> List[str]:
         # Not used by LocalWorkspace
@@ -335,7 +339,7 @@ def create_stack(stack_name: str,
     if _is_inline_program(**args):
         # Type checks are ignored because we have already asserted that the correct args are present.
         return _inline_source_stack_helper(stack_name, program, project_name, Stack.create, opts)  # type: ignore
-    elif _is_local_program(**args):
+    if _is_local_program(**args):
         return _local_source_stack_helper(stack_name, work_dir, Stack.create, opts)  # type: ignore
     raise ValueError(f"unexpected args: {' '.join(args)}")
 
@@ -373,7 +377,7 @@ def select_stack(stack_name: str,
     args = locals()
     if _is_inline_program(**args):
         return _inline_source_stack_helper(stack_name, program, project_name, Stack.select, opts)  # type: ignore
-    elif _is_local_program(**args):
+    if _is_local_program(**args):
         return _local_source_stack_helper(stack_name, work_dir, Stack.select, opts)  # type: ignore
     raise ValueError(f"unexpected args: {' '.join(args)}")
 
@@ -411,7 +415,7 @@ def create_or_select_stack(stack_name: str,
     args = locals()
     if _is_inline_program(**args):
         return _inline_source_stack_helper(stack_name, program, project_name, Stack.create_or_select, opts)  # type: ignore
-    elif _is_local_program(**args):
+    if _is_local_program(**args):
         return _local_source_stack_helper(stack_name, work_dir, Stack.create_or_select, opts)  # type: ignore
     raise ValueError(f"unexpected args: {' '.join(args)}")
 

--- a/sdk/python/lib/pulumi/x/automation/_project_settings.py
+++ b/sdk/python/lib/pulumi/x/automation/_project_settings.py
@@ -83,7 +83,7 @@ class ProjectSettings:
                  description: Optional[str] = None,
                  author: Optional[str] = None,
                  website: Optional[str] = None,
-                 license: Optional[str] = None,
+                 license: Optional[str] = None, # pylint: disable=redefined-builtin
                  config: Optional[str] = None,
                  template: Optional[ProjectTemplate] = None,
                  backend: Optional[ProjectBackend] = None):

--- a/sdk/python/lib/pulumi/x/automation/_server.py
+++ b/sdk/python/lib/pulumi/x/automation/_server.py
@@ -13,10 +13,10 @@
 # limitations under the License.
 
 import asyncio
-import grpc
 import sys
 import traceback
 from contextlib import suppress
+import grpc
 
 from ._workspace import PulumiFn
 from ... import log
@@ -70,6 +70,7 @@ class LanguageServer(LanguageRuntimeServicer):
         except grpc.RpcError as exn:
             # If the monitor is unavailable, it is in the process of shutting down or has already
             # shut down. Don't emit an error if this is the case.
+            # pylint: disable=no-member
             if exn.code() == grpc.StatusCode.UNAVAILABLE:
                 log.debug("Resource monitor has terminated, shutting down.")
             else:

--- a/sdk/python/lib/pulumi/x/automation/_stack.py
+++ b/sdk/python/lib/pulumi/x/automation/_stack.py
@@ -13,11 +13,11 @@
 # limitations under the License.
 
 import json
-import grpc
 from concurrent import futures
 from enum import Enum
 from datetime import datetime
 from typing import List, Any, Mapping, MutableMapping, Optional
+import grpc
 
 from ._cmd import CommandResult, _run_pulumi_cmd, OnOutput
 from ._config import ConfigValue, ConfigMap, _SECRET_SENTINEL
@@ -155,6 +155,8 @@ class DestroyResult(BaseResult):
 
 
 class Stack:
+    # pylint: disable=unused-argument
+
     name: str
     """The name identifying the Stack."""
 
@@ -278,9 +280,11 @@ class Stack:
             port = server.add_insecure_port(address="0.0.0.0:0")
             server.start()
 
-            def on_exit():
+            def on_exit_fn():
                 language_server.on_pulumi_exit()
                 server.stop(0)
+            on_exit = on_exit_fn
+
             args.append(f"--client=127.0.0.1:{port}")
 
         args.extend(["--exec-kind", kind])
@@ -289,7 +293,7 @@ class Stack:
             up_result = self._run_pulumi_cmd_sync(args, on_output)
             outputs = self.outputs()
             summary = self.info()
-            assert (summary is not None)
+            assert summary is not None
             return UpResult(stdout=up_result.stdout, stderr=up_result.stderr, summary=summary, outputs=outputs)
         finally:
             if on_exit is not None:
@@ -340,9 +344,11 @@ class Stack:
             port = server.add_insecure_port(address="0.0.0.0:0")
             server.start()
 
-            def on_exit():
+            def on_exit_fn():
                 language_server.on_pulumi_exit()
                 server.stop(0)
+            on_exit = on_exit_fn
+
             args.append(f"--client=127.0.0.1:{port}")
         args.extend(["--exec-kind", kind])
 
@@ -381,7 +387,7 @@ class Stack:
         self.workspace.select_stack(self.name)
         refresh_result = self._run_pulumi_cmd_sync(args, on_output)
         summary = self.info()
-        assert(summary is not None)
+        assert summary is not None
         return RefreshResult(stdout=refresh_result.stdout, stderr=refresh_result.stderr, summary=summary)
 
     def destroy(self,
@@ -411,7 +417,7 @@ class Stack:
         self.workspace.select_stack(self.name)
         destroy_result = self._run_pulumi_cmd_sync(args, on_output)
         summary = self.info()
-        assert(summary is not None)
+        assert summary is not None
         return DestroyResult(stdout=destroy_result.stdout, stderr=destroy_result.stderr, summary=summary)
 
     def get_config(self, key: str) -> ConfigValue:
@@ -529,7 +535,7 @@ class Stack:
         :returns: Optional[UpdateSummary]
         """
         history = self.history(page_size=1)
-        if not len(history):
+        if not history:
             return None
         return history[0]
 

--- a/sdk/python/lib/pulumi/x/automation/_stack_settings.py
+++ b/sdk/python/lib/pulumi/x/automation/_stack_settings.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Optional, Mapping, Any
+from typing import Optional, Mapping, Any, Union, Dict
 
 
 class StackSettingsSecureConfigValue:
@@ -39,10 +39,10 @@ class StackSettings:
         self.encrypted_key = encrypted_key
         self.encryption_salt = encryption_salt
         if config:
-            stack_config = {}
+            stack_config: Dict[str, Union[str, StackSettingsSecureConfigValue]] = {}
             for key in config:
                 val = config[key]
-                if type(val) == str:
+                if isinstance(val, str):
                     stack_config[key] = val
                 elif "secure" in val:
                     stack_config[key] = StackSettingsSecureConfigValue(**val)

--- a/sdk/python/lib/pulumi/x/automation/_workspace.py
+++ b/sdk/python/lib/pulumi/x/automation/_workspace.py
@@ -143,7 +143,6 @@ class Workspace(ABC):
 
         :returns: ProjectSettings
         """
-        pass
 
     @abstractmethod
     def save_project_settings(self, settings: ProjectSettings) -> None:
@@ -153,7 +152,6 @@ class Workspace(ABC):
 
         :param settings: The project settings to save.
         """
-        pass
 
     @abstractmethod
     def stack_settings(self, stack_name: str) -> StackSettings:
@@ -163,7 +161,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :return: StackSettings
         """
-        pass
 
     @abstractmethod
     def save_stack_settings(self, stack_name: str, settings: StackSettings) -> None:
@@ -173,7 +170,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :param settings: The stack settings to save.
         """
-        pass
 
     @abstractmethod
     def serialize_args_for_op(self, stack_name: str) -> List[str]:
@@ -184,7 +180,6 @@ class Workspace(ABC):
 
         :param stack_name: The name of the stack.
         """
-        pass
 
     @abstractmethod
     def post_command_callback(self, stack_name: str) -> None:
@@ -195,7 +190,6 @@ class Workspace(ABC):
 
         :param stack_name: The name of the stack.
         """
-        pass
 
     @abstractmethod
     def get_config(self, stack_name: str, key: str) -> ConfigValue:
@@ -207,7 +201,6 @@ class Workspace(ABC):
         :param key: The key for the config item to get.
         :returns: ConfigValue
         """
-        pass
 
     @abstractmethod
     def get_all_config(self, stack_name: str) -> ConfigMap:
@@ -217,7 +210,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :returns: ConfigMap
         """
-        pass
 
     @abstractmethod
     def set_config(self, stack_name: str, key: str, value: ConfigValue) -> None:
@@ -228,7 +220,6 @@ class Workspace(ABC):
         :param key: The config key to add.
         :param value: The config value to add.
         """
-        pass
 
     @abstractmethod
     def set_all_config(self, stack_name: str, config: ConfigMap) -> None:
@@ -238,7 +229,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :param config: A mapping of key to ConfigValue to set to config.
         """
-        pass
 
     @abstractmethod
     def remove_config(self, stack_name: str, key: str) -> None:
@@ -248,7 +238,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :param key: The key to remove from config.
         """
-        pass
 
     @abstractmethod
     def remove_all_config(self, stack_name: str, keys: List[str]) -> None:
@@ -258,7 +247,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack.
         :param keys: The keys to remove from config.
         """
-        pass
 
     @abstractmethod
     def refresh_config(self, stack_name: str) -> None:
@@ -267,7 +255,6 @@ class Workspace(ABC):
 
         :param stack_name: The name of the stack.
         """
-        pass
 
     @abstractmethod
     def who_am_i(self) -> WhoAmIResult:
@@ -276,7 +263,6 @@ class Workspace(ABC):
 
         :returns: WhoAmIResult
         """
-        pass
 
     @abstractmethod
     def stack(self) -> Optional[StackSummary]:
@@ -285,7 +271,6 @@ class Workspace(ABC):
 
         :returns: Optional[StackSummary]
         """
-        pass
 
     @abstractmethod
     def create_stack(self, stack_name: str) -> None:
@@ -296,7 +281,6 @@ class Workspace(ABC):
         :returns: None
         :raises CommandError Raised if a stack with the same name exists.
         """
-        pass
 
     @abstractmethod
     def select_stack(self, stack_name: str) -> None:
@@ -307,7 +291,6 @@ class Workspace(ABC):
         :returns: None
         :raises CommandError Raised if no matching stack exists.
         """
-        pass
 
     @abstractmethod
     def remove_stack(self, stack_name: str) -> None:
@@ -316,7 +299,6 @@ class Workspace(ABC):
 
         :param stack_name: The name of the stack to remove
         """
-        pass
 
     @abstractmethod
     def list_stacks(self) -> List[StackSummary]:
@@ -327,7 +309,6 @@ class Workspace(ABC):
 
         :returns: List[StackSummary]
         """
-        pass
 
     @abstractmethod
     def install_plugin(self, name: str, version: str, kind: str = "resource") -> None:
@@ -338,7 +319,6 @@ class Workspace(ABC):
         :param version: The version to install.
         :param kind: The kind of plugin.
         """
-        pass
 
     @abstractmethod
     def remove_plugin(self,
@@ -352,7 +332,6 @@ class Workspace(ABC):
         :param version_range: The version range to remove.
         :param kind: The kind of plugin.
         """
-        pass
 
     @abstractmethod
     def list_plugins(self) -> List[PluginInfo]:
@@ -361,7 +340,6 @@ class Workspace(ABC):
 
         :returns: List[PluginInfo]
         """
-        pass
 
     @abstractmethod
     def export_stack(self, stack_name: str) -> Deployment:
@@ -372,7 +350,6 @@ class Workspace(ABC):
         :param stack_name: The name of the stack to export.
         :returns: Deployment
         """
-        pass
 
     @abstractmethod
     def import_stack(self, stack_name: str, state: Deployment) -> None:
@@ -383,4 +360,3 @@ class Workspace(ABC):
         :param stack_name: The name of the stack to import.
         :param state: The deployment state to import.
         """
-        pass

--- a/sdk/python/lib/pulumi/x/automation/errors.py
+++ b/sdk/python/lib/pulumi/x/automation/errors.py
@@ -49,7 +49,7 @@ class InlineSourceRuntimeError(CommandError):
         self.name = "InlineSourceRuntimeError"
 
 
-class RuntimeError(CommandError):
+class RuntimeError(CommandError): # pylint: disable=redefined-builtin
     def __init__(self, command_result: 'CommandResult'):
         super().__init__(command_result)
         self.name = "RuntimeError"

